### ISSUE TITLE
Azure Functions Http: missing HTTP method definitions for delete and patch

### ISSUE THREAD

**Issue by Nithanim-Work** [quarkusio/quarkus#37065]
### Describe the bug

In the function definition for the generic http request handler, the supported HTTP methods are manually defined as list. Some obvious methods are listed, like `GET` or `POST`, but `DELETE` and `PATCH` are missing.

The function defined in [the Quarkus code](https://github.com/quarkusio/quarkus/blob/7f5029944a47906715a32addac682435147c7f5e/extensions/azure-functions-http/runtime/src/main/java/io/quarkus/azure/functions/resteasy/runtime/Function.java) looks like this:
```java
FunctionName(QUARKUS_HTTP)
    public HttpResponseMessage run(
            @HttpTrigger(name = "req", dataType = "binary", methods = { HttpMethod.GET, HttpMethod.HEAD, HttpMethod.POST,
                    HttpMethod.PUT,
                    HttpMethod.OPTIONS }, route = "{*path}", authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context)
```

Since `DELETE` and `PATCH` are missing, the Function Host (or whatever agent handling the request in azure) responds to any `DELETE` request with `404` directly because no route matches.

My current workaround patches the `functions.json` after Quarkus generated the function package:
```xml
             <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-antrun-plugin</artifactId>
                <version>3.0.0</version>
                <executions>
                    <execution>
                        <phase>package</phase>
                        <configuration>
                            <target>
                                <replace file="target/azure-functions/bin/QuarkusHttp/function.json" token="&quot;GET&quot;, &quot;HEAD&quot;" value="&quot;GET&quot;, &quot;HEAD&quot;, &quot;PATCH&quot;, &quot;DELETE&quot;" />
                            </target>
                        </configuration>
                        <goals>
                            <goal>run</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
```

Allegedly, you can leave the list empty to implicitly match all HTTP methods [as the documentation states in the "Attributes" section](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=python-v2%2Cin-process%2Cnodejs-v4%2Cfunctionsv2&pivots=programming-language-csharp#attributes) but I have not tested that.

### Expected behavior

HTTP request using the delete `DELETE` method reaches Quarkus and is executed normally.

### Actual behavior

Since the HTTP method `DELETE` does not match the function, the host returns 404 without indication to the user why.

**Comment by franden** [comment]
the issue is still present and is very annoying. @geoand could you please analyze the problem?

**Comment by geoand** [comment]
I have never worked with Azure Functions, sorry

**Comment by Nithanim-Work** [comment]
> I have never worked with Azure Functions, sorry

I am very happy for you!

> the issue is still present and is very annoying. @geoand could you please analyze the problem?

Sorry, I haven't found time to do it myself. I am just happy if anything works at all in this damn azure cloud. And my workaround was enough for me to move on to many other problems.

**Comment by geoand** [comment]
I suppose I could simply add the missing methods, but I have no way of testing it

**Comment by Nithanim-Work** [comment]
> I suppose I could simply add the missing methods, but I have no way of testing it

Yeah, that is also a bit of my problem. For other stuff, I broke out the extension in a custom one and applied fixes. But porting the changes back is a copy-paste action where I can make mistakes and then I can't really test.

But since the pressure is slowly winding down at my project at work, I might be able to port the fix for the encoding (#36973) back. I would YOLO that also since virtually nobody seems to use Quarkus with Azure Functions and we use it in production.
